toolkit/tools/pkg/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,411 @@
+	exceptions       LicenseExceptions                  // Files that should be ignored
+	results          []LicenseCheckResult               // The results of the search
+	jobSemaphore     chan struct{}                      // Limit the number of parallel jobs
+}
+
+// New creates a new license checker. If this returns successfully the caller is responsible for calling CleanUp().
+// - buildDirPath: The path to create the chroot inside
+// - workerTarPath: The path to the worker tarball
+// - rpmDirPath: The path to the directory containing the RPMs
+// - nameFilePath: The path to the .json file containing license names
+// - exceptionFilePath: Optional, the path to the .json file containing license exceptions to ignore
+// - distTag: The distribution tag to use when parsing RPMs
+func New(buildDirPath, workerTarPath, rpmDirPath, nameFilePath, exceptionFilePath, distTag string,
+) (newLicenseChecker *LicenseChecker, err error) {
+	newLicenseChecker = &LicenseChecker{
+		distTag:          distTag,
+		simpleToolChroot: &simpletoolchroot.SimpleToolChroot{},
+		jobSemaphore:     make(chan struct{}, numParallelJobs),
+	}
+
+	err = newLicenseChecker.simpleToolChroot.InitializeChroot(buildDirPath, chrootName, workerTarPath, rpmDirPath)
+	if err != nil {
+		newLicenseChecker.CleanUp()
+		err = fmt.Errorf("failed to initialize chroot:\n%w", err)
+		return nil, err
+	}
+
+	newLicenseChecker.licenseNames, err = LoadLicenseNames(nameFilePath)
+	if err != nil {
+		newLicenseChecker.CleanUp()
+		err = fmt.Errorf("failed to load license names:\n%w", err)
+		return nil, err
+	}
+
+	if exceptionFilePath != "" {
+		newLicenseChecker.exceptions, err = LoadLicenseExceptions(exceptionFilePath)
+		if err != nil {
+			newLicenseChecker.CleanUp()
+			err = fmt.Errorf("failed to load license exceptions:\n%w", err)
+			return nil, err
+		}
+	}
+
+	return newLicenseChecker, err
+}
+
+// CleanUp tears down the chroot. If the chroot was created it will be cleaned up. Reset the struct to its initial state.
+func (l *LicenseChecker) CleanUp() error {
+	if l.simpleToolChroot != nil {
+		err := l.simpleToolChroot.CleanUp()
+		if err != nil {
+			return fmt.Errorf("failed to cleanup chroot:\n%w", err)
+		}
+		l.simpleToolChroot = nil
+	}
+	return nil
+}
+
+// CheckLicenses will scan all .rpm files in the chroot for bad licenses. The results can be accessed with FormatResults() or GetAllResults().
+func (l *LicenseChecker) CheckLicenses() error {
+	if l.simpleToolChroot == nil {
+		return fmt.Errorf("chroot has not been initialized")
+	}
+
+	l.results = []LicenseCheckResult{}
+
+	err := l.simpleToolChroot.RunInChroot(func() (searchErr error) {
+		l.results, searchErr = l.runLicenseCheckInChroot()
+		return searchErr
+	})
+	if err != nil {
+		return fmt.Errorf("failed to scan for license issues:\n%w", err)
+	}
+
+	for _, result := range l.results {
+		if result.err != nil {
+			logger.Log.Errorf("Failed to scan (%s) for license issues: %v", result.RpmPath, result.err)
+		}
```

Comment:
Do we want to exit `CheckLicenses` without an error in this case?

Reply:
This is dead code, `l.runLicenseCheckInChroot()` will return an error if any worker does, so this can just go.